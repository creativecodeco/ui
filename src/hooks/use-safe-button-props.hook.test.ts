/* eslint-disable @typescript-eslint/no-explicit-any */
import { renderHook } from '@testing-library/react';

import useSafeButtonProps from './use-safe-button-props.hook';

describe('useSafeButtonProps', () => {
  it('disabled false should execute events', () => {
    const props = {
      onClick: jest.fn(),
      onSubmit: jest.fn(),
      disabled: false
    };
    const result = renderHook(() => useSafeButtonProps(props)).result.current;

    expect(result.disabled).toBe(false);

    result.onClick({} as any);
    result.onSubmit({} as any);

    expect(props.onClick).toHaveBeenCalledTimes(1);
    expect(props.onSubmit).toHaveBeenCalledTimes(1);
  });

  it('disabled trusty should not execute events', () => {
    const props = {
      onClick: jest.fn(),
      onSubmit: jest.fn(),
      disabled: true
    };
    const result = renderHook(() => useSafeButtonProps(props)).result.current;

    expect(result.disabled).toBe(true);

    result.onClick({} as any);
    result.onSubmit({} as any);

    expect(props.onClick).toHaveBeenCalledTimes(0);
    expect(props.onSubmit).toHaveBeenCalledTimes(0);
  });

  it('should return props', () => {
    const props = {
      onClick: jest.fn(),
      onSubmit: jest.fn(),
      disabled: false
    };

    const result = renderHook(() => useSafeButtonProps(props)).result.current;

    expect(result).toMatchSnapshot();
  });
});
