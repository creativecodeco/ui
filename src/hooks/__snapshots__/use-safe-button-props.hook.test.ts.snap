// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSafeButtonProps should return props 1`] = `
{
  "disabled": false,
  "onClick": [Function],
  "onSubmit": [Function],
}
`;
